components/ProgressView/examples/ProgressViewPillExample.m

```diff
@@ -0,0 +1,128 @@
+// Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+#import "MaterialContainerScheme.h"
+#import "MaterialPalettes.h"
+#import "MaterialProgressView.h"
+
+static const CGFloat kProgressViewHeight = 4;
+
+@interface ProgressViewPillExample : UIViewController
+
+@property(nonatomic, strong) UIView *container;
+
+@property(nonatomic, strong) MDCProgressView *progressView;
+@property(nonatomic, strong) UILabel *progressLabel;
+
+@property(nonatomic, strong) MDCContainerScheme *containerScheme;
+
+@end
+
+@implementation ProgressViewPillExample
+
+- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
+  self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
+  if (self) {
+    self.title = @"Progress View Pill";
+    self.navigationItem.rightBarButtonItem =
+        [[UIBarButtonItem alloc] initWithTitle:@"Advance"
+                                         style:UIBarButtonItemStylePlain
+                                        target:self
+                                        action:@selector(didPressAdvanceButton:)];
+    self.navigationItem.rightBarButtonItem.accessibilityIdentifier = @"advance_button";
+
+    self.containerScheme = [[MDCContainerScheme alloc] init];
+  }
+  return self;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  self.view.backgroundColor = self.containerScheme.colorScheme.backgroundColor;
+
+  self.container = [[UIView alloc] initWithFrame:self.view.bounds];
+  [self.view addSubview:self.container];
+
+  _progressView = [[MDCProgressView alloc] init];
+  _progressView.translatesAutoresizingMaskIntoConstraints = NO;
+  [self.container addSubview:_progressView];
+  _progressView.progress = (float)0.25;
+
+  // Configure the "pill" effect by giving both the container and the progress a corner radius.
+  _progressView.layer.cornerRadius = kProgressViewHeight / 2;
```

Comment:
I'm a bit confused here.
What is stopping the user from doing:
_progressView.layer.cornerRadius = kProgressViewHeight / 2
_progressView.clipsToBounds = YES

or if not, why do they have to set the corner radius twice? why not just:
_progressView.progressCornerRadius = kProgressViewHeight / 2

Reply:
The following only rounds the container of the progress view, not the progress bar itself:

```
_progressView.layer.cornerRadius = kProgressViewHeight / 2
_progressView.clipsToBounds = YES
```

We might be able to make the latter work...let me try something.

Reply:
Made an improvement to the API so that it configures both the progress view and the track view. This gives the client a one-liner for configuring the corner radius, while also avoiding conflicting with any values set to layer.cornerRadius on the progress view itself.

Reply:
Looks great!